pdk/1.0/modules/ROOT/pages/policies-pdk-configure-features-inject-parameters.adoc

```diff
@@ -1,28 +1,33 @@
-= Inject Parameters
+= Injecting Parameters
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images
 
-The default functions receive the predefined parameters:
+By default, the #[entrypoint] configuration function receives the following injectable parameters:
 
-* `#[entrypoint]` configuration function:
-** `Configuration`
-** `HttpClient`
-** `CacheBuilder`
+* `Configuration`: Provides the policy's configuration parameters. For information about how to define configuration parameters, see xref:policies-pdk-create-schema-definition.adoc[].
+* `Metadata`: Provides metadata about the Policy, the Flex instance, the API instance and the Anypoint Organization. For more information about the metadata provided, see <<metadata>>.
+* `HttpClient`: Enables the policy to make HTTP calls. For more information about how to make HTTP calls from the policy, see xref:policies-pdk-configure-features-http-request.adoc[].
+* `CacheBuilder`: Provides the caching features of the policy. For more information about caching, see xref:policies-pdk-configure-features-caching.adoc[].
+* `StreamProperties`: Provides a structure to share properties with other policies that process the same request. For more information about sharing information between policies, see <<streamproperties>>.
 
-* `on_request` wrapped functions:
-** `RequestState`
-** `HttpClient`
+You can also directly inject the `HttpClient` and `StreamProperties` parameter into the wrapped functions. Choose where to inject these parameters based on where they are used. For example, because the HTTP call must be performed inside the wrapped functions, it is best to directly inject. However, if you want to use the `HttpClient` in both `on_response` and `on_request` functions, you should inject the `HttpClient` in the `#[entrypoint]`. 
```

Comment:
Im not sure this paragraph actually helps. Unfortunately this topic is very technical. Let me know if this paragraph makes sense.

```
HTTPClient and StreamProperties also can be directly injected into a filter funtion, wrapped either by `on_request` or `on_response`. So, for instance, if you need to perform an HTTP call inside the `on_request` filter, the HTTPClient can be injected in that function.

As a rule of thumb, inject the parameters in the context where they must be used. The only exception for this rule is when we want to optimize the policy processing. It is always better to perform initializations in the entrypoint function whenever it is possible, which is executed when the policy is configured. This way we are saving that processing in filter function that are executed each time a request is processed.
```

What I am trying to say is:
- Use these utilities where you need them
- Use them in the entrypoint function when you need to perform some operations to initialize stuff. Making that operations in entrypoint function and not in filter functions will make your policy more efficient.